doc/sailfish-webview-webview.qdoc

```diff
@@ -14,4 +14,60 @@
   import namespace.
 */
 
+/*!
+  \qmlproperty var WebView::popupComponents
+  \brief Contains a dictionary of custom user interaction components
+
+  Client applications can define their own custom user interaction components
+  which extend the various interface types defined in the
+  \l{Sailfish.WebView.Popups} namespace.
+
+  In order for those custom components to be used by the WebView, the client
+  must then register those components with the WebView via this property.
+  The value is a dictionary whose keys are the asynchronous message types
+  which trigger the popup, and whose values are the \c{Component}s (or
+  component type definition URLs) which the WebView should load and display
+  when it receives the trigger message.
+
+  The dictionary keys are strings, and the values are either a custom
+  component entry object, or a popup-sub-type dictionary (whose key is
+  the sub-type name, and whose values are custom component entry objects).
+
+  The valid dictionary keys are as follows:
+  \list
+  \li \c{"Content:ContextMenu"} to trigger a \l{ContextMenuInterface}{context menu popup}
+  \li \c{"embed:alert"} to trigger an \l{AlertPopupInterface}{alert popup}
+  \li \c{"embed:confirm"} to trigger a \l{ConfirmPopupInterface}{confirm popup}
+  \li \c{"embed:prompt"} to trigger a \l{PromptPopupInterface}{input prompt popup}
+  \li \c{"embed:login"} to trigger an \l{AuthPopupInterface}{authorization/login popup}
+  \li \c{"embed:auth"} to trigger a \l{PasswordManagerPopupInterface}{password manager popup}
+  \li \c{"embed:permissions"} to trigger a permission-to-use popup of a specified subtype, identified by subkey.
```

Comment:
This could be part of the same enum as well.

GeoLocationPermission enum maybe. internal mapping to "embed:permissions" + subtype "geolocation".

Reply:
Yep, good point.  No need to push two-step resolution complexity to client.  I will add a specific string key for this permission type, and do the subtopic mapping manually in PopupOpener.qml